### ISSUE TITLE
playInBackground Ignored in iOS

### ISSUE THREAD

**Issue by brianlenz** [TheWidlarzGroup/react-native-video#2954]
# Bug
iOS devices seem to always play audio in the background, even when it's not enabled. Per the docs, [playInBackground](https://github.com/react-native-video/react-native-video/blob/master/API.md#playinbackground) should default to `false`, but even if you explicitly set `playInBackground={false}`, the audio will continue to play when the app is backgrounded, which is not the expected behavior.

Is this a known issue by chance?

## Platform
Which player are you experiencing the problem on:
* iOS

## Environment info
React native info output:

```bash
System:
    OS: macOS 12.6.1
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Memory: 51.96 MB / 16.00 GB
    Shell: 5.2.9 - /usr/local/bin/bash
  Binaries:
    Node: 19.1.0 - /private/var/folders/fq/d3nyh22d32j20vxw03_hl5gh0000gn/T/xfs-0df7992b/node
    Yarn: 3.2.2 - /private/var/folders/fq/d3nyh22d32j20vxw03_hl5gh0000gn/T/xfs-0df7992b/yarn
    npm: 8.19.3 - /usr/local/bin/npm
    Watchman: 2022.11.14.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /Users/brian/.rbenv/shims/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.1, iOS 16.1, macOS 13.0, tvOS 16.1, watchOS 9.1
    Android SDK:
      API Levels: 25, 29, 30, 31, 33
      Build Tools: 26.0.2, 26.0.3, 28.0.3, 29.0.2, 30.0.2, 30.0.3, 31.0.0, 31.0.0, 31.0.0, 33.0.0
      System Images: android-30 | Google APIs Intel x86 Atom, android-31 | Google APIs Intel x86 Atom_64
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.3 AI-213.7172.25.2113.9014738
    Xcode: 14.1/14B47b - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0 
    react-native: 0.70.6 => 0.70.6 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version: `6.0.0-alpha.4`

## Steps To Reproduce
1. While the video is playing, background the app.

## Expected behaviour
1. Video pauses while app is in background.

## Reproducible sample code
Very simple repro:
```jsx
<Video
  key={videoURI}
  source={{
    uri: videoURI,
  }}
  rate={1.0}
  volume={1.0}
  muted={false}
  paused={false}
  resizeMode="contain"
  controls={false}
  repeat={true}
  playInBackground={false}
  onEnd={() => {
    setPaused(true);
  }}
/>
```

**Comment by AhmedAbuelenin** [comment]
I can confirm that playInBackground in iOS is working as expected video is stopped in background but when app is in foreground again it doesn't run and the same thing is happened with playWhenInactive. I am using default values for both.
 I am using the latest version of the lib v6.0.0-alpha.4
 react-native: 0.70.6

**Comment by brianlenz** [comment]
@AhmedAbuelenin hmm. I wonder what you are doing differently, as it's definitely not working for us on `6.0.0-alpha.4`. When the app is backgrounded, the audio skips briefly, but then it keeps playing while the app is in the background.

**Comment by AhmedAbuelenin** [comment]
@brianlenz 
I was passing rate={1.0} to component but after deleting it I experienced the same problem like yours in simulator 16.1 but it worked just fine on my device 16.1 very weird!!

**Comment by jstheoriginal** [comment]
For the time being, you can use a hook like this, and then set play and mute to true always if the app is not in the foreground (active).

```
/**
 * Returns the app's current "app state" value (eg. 'active', 'inactive', 'background', etc.).
 */
function useAppState() {
  const [appState, setAppState] = useState<AppStateStatus>(AppState.currentState)

  useEffect(() => {
    function onChange(newState: AppStateStatus) {
      setAppState(newState)
    }

    const subscription = AppState.addEventListener('change', onChange)

    return () => {
      subscription.remove()
    }
  }, [])

  return appState
}
```

then:

```
// you can also add `'inactive'` if you want it to keep playing when the notification/control centers are active)
const canPlay = useAppState() === 'active'

...
// (assuming you're controlling playing and mute with useState somewhere)
// muting isn't probably necessary if paused, but I just added it so it's explict

pause={canPlay ? !playing : true}
mute={canPlay ? mute : true}
```

This is working for me to get around the issue for now.

**Comment by brianlenz** [comment]
Thanks, @jstheoriginal! Brilliant 🚀

**Comment by freeboub** [comment]
A fix will be available in 6.0.0-beta.7.
Please confirm the fix on this release !
